### ISSUE TITLE
request:show progress bar within the icon

### ISSUE THREAD

**Issue by laurentopia** [ForkIssues/TrackerWin#1250]
Long commit require frequent click on git-fork to see where it's at.
All softwares I use show progress within the icon.
Like so.
![image](https://user-images.githubusercontent.com/26075468/126839309-ecb9d3b3-3bab-4cce-b172-dbccd1d807cb.png)

**Comment by DanPristupov** [comment]
`git commit` doesn't report progress: https://git-scm.com/docs/git-commit

**Comment by laurentopia** [comment]
git-fork shows a progress bar, probably based on the various steps such as compressing objects etc... if you know the number of steps required in a  git operation then you can display a progress bar as a fraction of the number of steps. not accurate but better than toggling to the window

**Comment by DanPristupov** [comment]
> if you know the number of steps required in a git operation

Well, there is just 1 step in commit.

> git-fork shows a progress bar, probably based on the various steps such as compressing objects etc...

No, for operations without progress (like commit) it's 5% and then immediately 100% when it's done. For operations with progress (fetch, push, etc), the progress shows real data. Not sure if we should show dummy progress globally.

**Comment by laurentopia** [comment]
> Not sure if we should show dummy progress globally.

It would help at least knowing that the operation is done.
Half green when started then no green when done gives a solid indication.

**Comment by laurentopia** [comment]
btw where do you pull all this info, it's very useful on large asset commit
![image](https://user-images.githubusercontent.com/26075468/126883813-8947efd5-df6b-4694-baef-403d6cbc0bed.png)

**Comment by DanPristupov** [comment]
`git push --progress` reports the progress status.

**Comment by AirQuick** [comment]
FWIW: When Microsoft Edge is downloading a file whose size is unknown, its taskbar icon continuously tells the download is still in progress.

https://user-images.githubusercontent.com/8489367/126983153-323176d3-9587-4723-ad5d-9fa6a6b54079.mp4

**Comment by laurentopia** [comment]
> `git push --progress` reports the progress status.

and do you know ahead of time how many steps the progress will go through?
anyway half way full then empty will be great enough, no need for precise progress bar

**Comment by DanPristupov** [comment]
> and do you know ahead of time how many steps the progress will go through?

Yes, I know the number of steps during `push`. But I don't see how it's related to `commit`.

**Comment by laurentopia** [comment]
> Yes, I know the number of steps during `push`. But I don't see how it's related to `commit`.

It's not, the topic is still "showing progress during a push"

**Comment by DanPristupov** [comment]
May be I'm missing something but you asked about commit.
<img width="400" alt="Screenshot 2021-08-04 at 08 31 41" src="https://user-images.githubusercontent.com/618115/128132904-323c5729-6e46-4cd9-b84d-b55ede76fb83.png">

**Comment by laurentopia** [comment]
Maybe, it's the same for me: putting things in a repo to protect from damage.
![image](https://user-images.githubusercontent.com/26075468/128269567-fc94401e-7896-4c6e-8e0d-598826364e65.png)
